includes/gateways/cod/class-wc-gateway-cod.php

```diff
@@ -208,6 +214,27 @@ public function is_available() {
 		return parent::is_available();
 	}
 
+	/**
+	 * Checks to see whether or not the admin settings are being accessed by the current request.
+	 *
+	 * @return bool
+	 */
+	private function is_accessing_settings() {
+		$screen = get_current_screen();
+		if ( $screen && $screen->in_admin() && 'woocommerce_page_wc-settings' === $screen->id ) {
```

Comment:
maybe I am missing something but this will be true for all pages in settings navigation right? I tested by adding a breakpoint here and looks like we return true on all WooCommerce settings pages and not just for Payments or COD page.

Perhaps we should check for `&tab=checkout` query param?

Reply:
In a previous commit I was actually doing this :D I've gone ahead and re-added that.